crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3357,8 +3357,16 @@ impl<'db> TypeInferenceBuilder<'db> {
             // When interacting with Todo, Any and Unknown should propagate (as if we fix this
             // `Todo` in the future, the result would then become Any or Unknown, respectively.)
             (Type::Any, _, _) | (_, Type::Any, _) => Some(Type::Any),
+            (todo @ Type::Todo(_), _, _) | (_, todo @ Type::Todo(_), _) => Some(todo),
+            (Type::Never, _, _) | (_, Type::Never, _) => Some(Type::Never),
             (Type::Unknown, _, _) | (_, Type::Unknown, _) => Some(Type::Unknown),
 
+            (Type::ClassLiteral(_), Type::ClassLiteral(_), ast::Operator::BitOr)
+                if Program::get(self.db()).python_version(self.db()) >= PythonVersion::PY310 =>
+            {
+                Some(UnionType::from_elements(self.db(), [left_ty, right_ty]))
+            }
+
```

Comment:
This is the clause that I think we don't want; I expect typeshed will natively give us the correct type of `types.UnionType`?

Reply:
Removed by the revert mentioned above